nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/cosmos/document/AbstractCosmosDocumentProcessor.java

```diff
@@ -0,0 +1,329 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.nifi.processors.azure.cosmos.document;
+
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import com.azure.cosmos.ConnectionPolicy;
+import com.azure.cosmos.ConsistencyLevel;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosClientException;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosDatabase;
+import com.azure.cosmos.models.CosmosContainerProperties;
+
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.annotation.lifecycle.OnStopped;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.services.azure.cosmos.AzureCosmosDocumentConnectionService;
+
+
+public abstract class AbstractCosmosDocumentProcessor extends AbstractProcessor {
+    static final String CONSISTENCY_STRONG = "STRONG";
+    static final String CONSISTENCY_BOUNDED_STALENESS= "BOUNDED_STALENESS";
+    static final String CONSISTENCY_SESSION = "SESSION";
+    static final String CONSISTENCY_CONSISTENT_PREFIX = "CONSISTENT_PREFIX";
+    static final String CONSISTENCY_EVENTUAL = "EVENTUAL";
+
+    static final Relationship REL_SUCCESS = new Relationship.Builder().name("success")
+    .description("All FlowFiles that are written to CosmoDB are routed to this relationship").build();
+
+    static final Relationship REL_FAILURE = new Relationship.Builder().name("failure")
+    .description("All FlowFiles that cannot be written to CosmoDB are routed to this relationship").build();
+
+    static final Relationship REL_ORIGINAL = new Relationship.Builder()
+            .name("original")
+            .description("All input FlowFiles that are part of a successful query execution go here.")
+            .build();
+
+    static final PropertyDescriptor CONNECTION_SERVICE = new PropertyDescriptor.Builder()
+        .name("Cosmos-Document-DB-Connection-Service")
+        .displayName("Cosmos Document DB Connection Service")
+        .description("If configured, this property will use the assigned for retrieving connection string info.")
+        .required(false)
+        .identifiesControllerService(AzureCosmosDocumentConnectionService.class)
+        .build();
+
+
+    static final PropertyDescriptor URI = new PropertyDescriptor.Builder()
+        .name("Cosmos Document DB URI")
```

Comment:
Could you make is consistent with other property names? Preferably lower case with - for example, cosmos-document-uri

Reply:
Done in AbstractAzureCosmosDBProcessor and AzureCosmosDBConnectionCtonrollerService. Note the changes of java files.